src/teleop/jetson/arm_trajectory_server.py

```diff
@@ -30,6 +31,41 @@ def joint_states_callback(msg: JointState):
         joint_states = msg
 
 
+def euclidean_error(threshold: int, feedback: FollowJointTrajectoryFeedback) -> Tuple[bool, str]:
+    position_errors = np.array(feedback.error.positions)
+    error = np.linalg.norm(position_errors)
+
+    if error > threshold:
+        error_msg = f"Euclidean error of {error} exceeded threshold of {threshold}"
+        return True, error_msg
+    return False, ""
+
+
+def joint_error(thresholds: list, feedback: FollowJointTrajectoryFeedback) -> Tuple[bool, str]:
+    position_errors = feedback.error.positions
+
+    for i in range(len(thresholds)):
+        if position_errors[i] > thresholds[i]:
+            error_msg = f"""
+                Joint {chr(i+65)} exceeded error threshold of {thresholds[i]}
+                Expected: {feedback.desired.positions[i]} rad")
+                Actual: {feedback.actual.positions[i]} rad")
+            """
+            return True, error_msg
+    return False, ""
+
+
+# Return true if arm has exceeded the error thresholds
+def error_threshold_exceeded(feedback: FollowJointTrajectoryFeedback) -> Tuple[bool, str]:
+    euclidean_error_threshold = rospy.get_param("teleop/euclidean_error_threshold")
+    joint_error_thresholds = list(rospy.get_param("teleop/joint_error_thresholds").values())
```

Comment:
I would be careful doing it this way, the order that you retrieve the params or the order they get put into a list might not be consistent. I think it would be alot safer to just access the values using the keys for the dict.